plugins/woocommerce/src/StoreApi/Schemas/V1/CheckoutSchema.php

```diff
@@ -261,20 +261,22 @@ function ( $key, $value ) {
 	 * @return array
 	 */
 	protected function get_additional_fields_response( \WC_Order $order ) {
-		$fields   = wp_parse_args(
-			$this->additional_fields_controller->get_all_fields_from_order( $order ),
-			$this->additional_fields_controller->get_all_fields_from_customer( wc()->customer )
+		$fields = wp_parse_args(
+			$this->additional_fields_controller->get_all_fields_from_object( $order, 'additional' ),
+			$this->additional_fields_controller->get_all_fields_from_object( wc()->customer, 'additional' )
 		);
-		$response = [];
 
+		$additional_field_schema = $this->get_additional_fields_schema();
 		foreach ( $fields as $key => $value ) {
-			if ( 0 === strpos( $key, '/billing/' ) || 0 === strpos( $key, '/shipping/' ) ) {
-				continue;
+			// This makes sure we're casting checkboxes from "1" and "0" to boolean. In the frontend, "0" is treated as truthy.
+			if ( isset( $additional_field_schema[ $key ]['type'] ) && 'boolean' === $additional_field_schema[ $key ]['type'] ) {
```

Comment:
I wonder if we also should check `isset( $additional_field_schema[ $key ] )` just to be safer.